Library/Homebrew/requirements.rb

```diff
@@ -126,3 +131,9 @@ class GitRequirement < Requirement
   default_formula "git"
   satisfy { Utils.git_available? }
 end
+
+class SubversionRequirement < Requirement
+  fatal true
+  default_formula "subversion"
```

Comment:
This isn't going to work as it's going to mean that bottled formulae on macOS always use the Subversion formula rather than the system version.

Reply:
Also, this probably warrants version check to make sure that the Subversion client is e.g. 1.8 or higher.

Reply:
> This isn't going to work as it's going to mean that bottled formulae on macOS always use the Subversion formula rather than the system version.

Since it's a `:build` dependency, bottled formula will not require the subversion formula.

> Also, this probably warrants version check to make sure that the Subversion client is e.g. 1.8 or higher.

I'll add the version check.

Reply:
We don't want to be pouring the subversion bottle while it's building either.

Reply:
I've added the version check.

Reply:
> We don't want to be pouring the subversion bottle while it's building either.

`brew install -s foo` will not pour the subversion bottle. `brew install --build-bottle foo` will pour the subversion bottle. As far as I can tell, there are only two formulae that use `svn://` for their stable URL: `smpeg` and `smpeg2`. So I don't think it's a big concern in practice. If it's something that you'd like to fix all-the-same, I can create a `SubversionRequirement` for macOS in `extend/os/mac/requirements.rb` that avoids pulling in a `subversion` dependency on macOS for `svn://`.

```sh
grep 'url.*svn:.*:revision' -r $(brew --repo homebrew/core)
./smpeg2.rb:  url "svn://svn.icculus.org/smpeg/tags/release_2_0_0/", :revision => "408"
./smpeg.rb:  url "svn://svn.icculus.org/smpeg/tags/release_0_4_5/", :revision => "399"
```

Reply:
This thing should be a no-op on macOS as far as I'm concerned.

Reply:
Another option if it's palatable to you is…
```ruby
default_formula "subversion" unless OS.mac?
```

> This thing should be a no-op on macOS as far as I'm concerned.

Cool. In that case, on macOS I'll make it…
```ruby
class SubversionRequirement < Requirement
  satisfy { true }
end
```

Reply:
Do you prefe

1. in the same file
```ruby
satisfy {
  next true if OS.mac?
  …
}
```

2. in extend/os/mac/requirements.rb
```ruby
satisfy { true }
```

Reply:
Now that I'm looking at both, I prefer option `extend/os/mac/requirements.rb` myself.

Reply:
> Now that I'm looking at both, I prefer option extend/os/mac/requirements.rb myself.

agreed

Reply:
Done.